src/py/reactpy/reactpy/core/types.py

```diff
@@ -73,6 +73,9 @@ class LayoutType(Protocol[_Render, _Event]):
     async def render(self) -> _Render:
         """Render an update to a view"""
 
+    async def renders(self) -> AsyncIterator[_Render]:
+        """Render a series of updates to a view"""
+
```

Comment:
Why not have `AsyncIterator` be an accepted type on `renders`, and then handle detection of this within `layout.py`?

Less APIs is generally more intuitive than more.